torch/distributed/_tensor/api.py

```diff
@@ -107,6 +107,15 @@ def backward(ctx, grad_output: torch.Tensor):  # type: ignore[override]
             None,
         )
 
+def allgather_metadata(local_tensor):
+    # Prepare local metadata
+    local_metadata = {'dtype': local_tensor.dtype, 'requires_grad': local_tensor.requires_grad}
```

Comment:
these three fields are not enough, we also need to include local tensor's shape/stride IF shape/stride are not passed in by user when calling `from_local`, we can add a `check_shape_stride` flag to `check_tensor_meta` call to include these two fields.